arangod/Aql/ExecutionBlock.h

```diff
@@ -99,7 +79,7 @@ class ExecutionBlock {
   /// if it returns an actual block, it must contain at least one item.
   /// getSome() also takes care of tracing and clearing registers; don't do it
   /// in getOrSkipSome() implementations.
-  virtual std::pair<ExecutionState, std::unique_ptr<AqlItemBlock>> getSome(size_t atMost);
+  virtual std::pair<ExecutionState, std::unique_ptr<AqlItemBlock>> getSome(size_t atMost) = 0;
 
   void traceGetSomeBegin(size_t atMost);
   void traceGetSomeEnd(AqlItemBlock const*, ExecutionState state);
```

Comment:
While these are implemented here, there should be no need to have any similar named methods in ExecutionBlockImpl or the cluster blocks. I'd suggest to switch the argument order here like this:
```suggestion
  void traceGetSomeEnd(ExecutionState state, AqlItemBlock const*);
```
so it matches the result of `getSome`. Or even take a pair so the result can just be passed.